src/execution_plan/optimizations/optimizer.c

```diff
@@ -54,6 +46,24 @@ void optimizePlan(ExecutionPlan *plan) {
 
 	// try to reduce execution plan incase it perform node or edge counting
 	reduceCount(plan);
+}
+
+// apply runtime optimizations
+void Optimizer_RuntimeOptimize
+(
+	ExecutionPlan *plan  // plan to optimize
+) {
+	// when possible, replace label scan and filter ops with index scans
+	utilizeIndices(plan);
+
+	// scan label with least entities
+	optimizeLabelScan(plan);
+
+	// try to reduce SCAN + FILTER to a node seek operation
+	seekByID(plan);
```

Comment:
can be moved to compile time

Reply:
Indeed, but this will require additional work, not sure it is worth investing in at the moment.